dulwich/porcelain.py

```diff
@@ -1189,9 +1193,12 @@ def check_ignore(repo, paths, no_index=False):
         ignore_manager = IgnoreFilterManager.from_repo(r)
         for path in paths:
             if os.path.isabs(path):
-                path = os.path.relpath(path, r.path)
```

Comment:
This is intentional. paths specified here are relative to the current path, not to the repository root (similar to the behaviour of "git check-ignore").

Reply:
hmm let me get this straight. 
- if at the start both `path` and `r.path` are already relative to the current path then you do not need this
- if at the start you know for sure that `r.path` is relative to the current path, and `path` might be absolute, then the right code should be: `path = os.path.relpath(path, os.getcwd())` no ?

Reply:
Either path or r.path can be absolute paths, so we do need the call to os.path.relpath. 

os.path.relpath will determine the absolute paths for path and r.path relative to os.getcwd() *and* then determine the relative path between them. E.g.:

    % cd /tmp
    /tmp% python
    Python 2.7.15 (default, May  1 2018, 05:55:50) 
    [GCC 7.3.0] on linux2
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import os
    >>> os.path.relpath("foo", "/home/blah/bar")
    '../../../tmp/foo'
    >>> os.path.relpath("/tmp/foo", "/home/blah/bar")
    '../../../tmp/foo'

"foo" gets interpreted as "/tmp/foo" and then the relative path from /home/blah/bar is determined.